### DISCUSSION TITLE
Bun v0.5.1

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#1857] [Announcements]
This release is a follow up to [Bun v0.5](https://github.com/oven-sh/bun/releases/tag/bun-v0.5.0)

To install:
```bash
curl https://bun.sh/install | bash
```

To upgrade:
```bash
bun upgrade
```

**[Read Bun v0.5.1's release notes on Bun's blog](https://bun.sh/blog/bun-v0.5.1)**

<a href="https://bun.sh/blog/bun-v0.5.1">
<img width="356" alt="image" src="https://user-images.githubusercontent.com/709451/213629472-51e98684-8f62-4b2e-8f39-076a92feb410.png">
</a>

## What's Changed
* Fix/simplify `bun-types` release by @colinhacks in https://github.com/oven-sh/bun/pull/1836
* add `fs.rmdir` & friends by @alexlamsl in https://github.com/oven-sh/bun/pull/1838
* support npm dependency aliasing by @alexlamsl in https://github.com/oven-sh/bun/pull/1837
* fix memory reference issues by @alexlamsl in https://github.com/oven-sh/bun/pull/1841
* fix stale references upon `Lockfile` tree cloning by @alexlamsl in https://github.com/oven-sh/bun/pull/1845
* parse dependency specifier correctly by @alexlamsl in https://github.com/oven-sh/bun/pull/1840
* fix(fetch:HTTP_PROXY) fix support for HTTP_PROXY/HTTPS_PROXY and NO_PROXY in fetch instances by @cirospaciari in https://github.com/oven-sh/bun/pull/1844
* Create new example http-file-extended.ts by @gornostay25 in https://github.com/oven-sh/bun/pull/1827
* repopulate `alias_map` correctly by @alexlamsl in https://github.com/oven-sh/bun/pull/1847
* Bugfixes to install by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/1848
* use `String.from()` by @alexlamsl in https://github.com/oven-sh/bun/pull/1850
* place `super` before generated code by @dylan-conway in https://github.com/oven-sh/bun/pull/1856

## New Contributors
* @gornostay25 made their first contribution in https://github.com/oven-sh/bun/pull/1827

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.5.0...bun-v0.5.1

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v0.5.1'>Bun v0.5.1</a>.</em>

**Comment by walterdiazesa** [comment]
This is the first time when `const { heapStats } = require('bun:jsc');` is available, but I cannot find it implemented in the changelog nor documented in the blog. Is it because I should be using heapStats differently prior to this version? In case it wasn't implemented, could it be helpful to add the inclusion to the [list](https://bun.sh/docs/project/benchmarking#javascript-heap-stats) of changelogs of this version?
